sdk/typescript/src/client/cryptography.ts

```diff
@@ -0,0 +1,358 @@
+ */
+export type PreparedSecondRound = {
+	/** The centralized public key share along with its zero-knowledge proof */
+	centralizedPublicKeyShareAndProof: Uint8Array;
+	/** The centralized public output from the DKG process */
+	centralizedPublicOutput: Uint8Array;
+	/** The centralized secret key share (should be encrypted before storage) */
+	centralizedSecretKeyShare: Uint8Array;
+	/** The encrypted user share with its proof of correct encryption */
+	encryptedUserShareAndProof: Uint8Array;
+};
+
+/**
+ * Prepared data for importing an existing cryptographic key as a DWallet.
+ * Contains verification data needed to prove ownership of the imported key.
+ */
+export type PreparedImportDWalletVerification = {
+	/** The secret share derived from the imported key */
+	secret_share: Uint8Array;
+	/** The public output that can be verified against the imported key */
+	public_output: Uint8Array;
+	/** The outgoing message for the verification protocol */
+	outgoing_message: Uint8Array;
+	/** The encrypted user share with proof for the imported key */
+	encryptedUserShareAndProof: Uint8Array;
+};
+
+/**
+ * Create a class groups keypair from a seed for encryption/decryption operations.
+ * Uses SECP256k1 curve with class groups for homomorphic encryption capabilities.
+ *
+ * @param seed - The seed bytes to generate the keypair from
+ * @returns Object containing the encryption key (public) and decryption key (private)
+ */
+export function createClassGroupsKeypair(seed: Uint8Array): {
+	encryptionKey: Uint8Array;
+	decryptionKey: Uint8Array;
+} {
+	const [expectedEncryptionKey, decryptionKey] = generate_secp_cg_keypair_from_seed(seed);
+
+	return {
+		encryptionKey: Uint8Array.from(expectedEncryptionKey),
+		decryptionKey: Uint8Array.from(decryptionKey),
+	};
+}
+
+/**
+ * Create the centralized output for the Distributed Key Generation (DKG) process.
+ * This function takes the first round output and produces the centralized party's contribution.
+ *
+ * @param networkDecryptionKeyPublicOutput - The network's public parameters for decryption
+ * @param firstRoundOutput - The output from the first round of DKG
+ * @param sessionIdentifier - Unique identifier for this DKG session
+ * @returns Object containing the centralized public key share with proof, public output, and secret key share
+ */
+export function createDKGCentralizedOutput(
+	networkDecryptionKeyPublicOutput: Uint8Array,
+	firstRoundOutput: Uint8Array,
+	sessionIdentifier: Uint8Array,
+): {
+	centralizedPublicKeyShareAndProof: Uint8Array;
+	centralizedPublicOutput: Uint8Array;
+	centralizedSecretKeyShare: Uint8Array;
+} {
+	const [centralizedPublicKeyShareAndProof, centralizedPublicOutput, centralizedSecretKeyShare] =
+		create_dkg_centralized_output(
+			networkDecryptionKeyPublicOutput,
+			Uint8Array.from(firstRoundOutput),
+			sessionIdentifierDigest(sessionIdentifier),
+		);
+
+	return {
+		centralizedPublicKeyShareAndProof: Uint8Array.from(centralizedPublicKeyShareAndProof),
```

Comment:
Note: this is the message to be sent to the network. If you want to abstract the cryptographic detail away from the users of this sdk you can rename it `centralizedPartyDKGMessage` or `userDKGMessage`